api/src/services/graphql.ts

```diff
@@ -112,6 +120,21 @@ export const GraphQLDate = new GraphQLScalarType({
 	description: 'ISO8601 Date values',
 });
 
+export const FilterStrNumber = new GraphQLScalarType({
+	name: 'FilterStrNumber',
+	serialize: stringNumber,
+	parseValue: stringNumber,
+	parseLiteral(ast) {
+		if (ast.kind === Kind.INT || ast.kind === Kind.FLOAT) {
+			return stringNumber(Number(ast.kind));
```

Comment:
Should this be `ast.value` instead of `ast.kind`?

```suggestion
			return stringNumber(Number(ast.value));
```

If that's the case, I'm also not entirely sure if `stringNumber` has the desired effect. Passing it `Number('rijk')` already passes `NaN` as the value, which means the output of `stringNumber(Number('rijk'))` will be `"NaN"` rather than the (presumably expected) `"rijk"`.

Reply:
@jaycammarano

Reply:
https://github.com/directus/directus/blob/74506f98312d525c2a1cde963ddbf8b750200e4f/api/src/services/graphql.ts#L127-L135